### PR TITLE
storeapi: remove dependency on the 'success' attr

### PR DESCRIPTION
**Pull request by sergiusens**
LP: #1615786

Signed-off-by: Sergio Schvezov <sergio.schvezov@ubuntu.com>

### CODE REVIEW

**Comment by sergiusens** [comment]
The work done here fixes LP: #1611116 as well

**Comment by sergiusens** [comment]
retest this please

File: snapcraft/tests/fake_servers.py [hunk 1]
```diff
@@ -235,14 +235,18 @@ def do_POST(self):
 
     def _handle_upload_request(self):
         logger.info('Handling upload request')
-        self.send_response(200)
-        self.send_header('Content-Type', 'application/octet-stream')
+        if 'UPDOWN_BROKEN' in os.environ:
```
Comment (kyrofa) [inline snapcraft/tests/fake_servers.py]: Why do this via environment variables instead of setting something on the fixture?

**Comment by kyrofa** [comment]
This looks pretty good, one question there.

File: snapcraft/tests/fake_servers.py (hunk 1 above)
Comment (sergiusens) [inline snapcraft/tests/fake_servers.py]: yeah, bug for @elopio to be able to change the behavior of the fixture

Comment (kyrofa) [inline snapcraft/tests/fake_servers.py]: Done, LP: [#1616117](https://bugs.launchpad.net/snapcraft/+bug/1616117).

**Comment by kyrofa** [comment]
:+1: from me.

**Pull request merged**